specs/lightclient/sync-protocol.md

```diff
@@ -0,0 +1,153 @@
+# Minimal Light Client
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Introduction](#introduction)
+- [Constants](#constants)
+- [Configuration](#configuration)
+  - [Misc](#misc)
+  - [Time parameters](#time-parameters)
+- [Containers](#containers)
+    - [`LightClientSnapshot`](#lightclientsnapshot)
+    - [`LightClientUpdate`](#lightclientupdate)
+    - [`LightClientStore`](#lightclientstore)
+- [Light client state updates](#light-client-state-updates)
+    - [`is_valid_light_client_update`](#is_valid_light_client_update)
+    - [`process_light_client_update`](#process_light_client_update)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+Eth2 is designed to be light client friendly for constrained environments to access Eth2 with reasonable satefy and liveness. Such environments include resource-constrained devices (e.g. phones for trust-minimised wallets) and metered VMs (e.g. blockchain VMs for cross-chain bridges).
+
+This document suggests a minimal light client design for the beacon chain that uses sync committees introduced in [this beacon chain extension](./beacon-chain.md).
+
+## Constants
+
+| Name | Value |
+| - | - |
+| `NEXT_SYNC_COMMITTEE_INDEX` | `IndexConcat(Index(BeaconBlock, 'state_root'), Index(BeaconState, 'next_sync_committee'))` |
+
+## Configuration
+
+### Misc
+
+| Name | Value |
+| - | - |
+| `MIN_SYNC_COMMITTEE_PARTICIPANTS` | `1` |
+| `MAX_VALID_LIGHT_CLIENT_UPDATES` | `uint64(2**64 - 1)` |
```

Comment:
we could be more precise with this bound as we should never have more than `LIGHT_CLIENT_UPDATE_TIMEOUT` updates queued under the scheme given below

Reply:
Why? `LIGHT_CLIENT_UPDATE_TIMEOUT` is denominated in slots whereas `MAX_VALID_LIGHT_CLIENT_UPDATES` is denominated in updates. There could be more than one update per slot! Remember these updates are prepared offchain. The onchain data is solely for incentivisation purposes.

Reply:
right -- i was thinking you would have at most one update per slot but this is not necessarily true, e.g. even getting the same update with simply more signatures attached